src/App.tsx

```diff
@@ -1,26 +1,105 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {useState} from 'react';
+
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+interface Product {
+    category: string;
+    price: string;
+    stocked: boolean;
+    name: string;
+}
+
+interface AllProducts {
+    products: Array<Product>;
+};
+
+export const App: React.FC<AllProducts> = ({products}) => {
+    const [searchText, setText] = useState("")
+    const [isOnlyStoked, setOnlyStocked] = useState(false)
+
+    const textProperties = {searchText, setText, isOnlyStoked, setOnlyStocked}
+
+    return (
+        <>
+            <SearchBar textProperties={textProperties}/>
+            <ProductTable products={products} isOnlyStoked={isOnlyStoked} searchText={searchText}/>
+        </>
+    )
+}
+const ProductRow: React.FC<{ product: Product }> = (product) => {
```

Comment:
// GOOD
```jsx
const ProductRow: React.FC<{ product: Product }> = (props) => {...}
```
// BAD
```jsx
const ProductRow: React.FC<{ product: Product }> = (thisIsWhatIsPassedToComponent) => {...}
```
---
Please update it to
```jsx
const ProductRow: React.FC<{ product: Product }> = (props) => {
const name = props.product...
}
```
OR:
```jsx
```jsx
// https://www.javatpoint.com/es6-object-destructuring
const ProductRow: React.FC<{ product: Product }> = ({product}) => {
const name = product...
}
```
```